publish/src/main/java/com/indix/gocd/s3publish/PublishExecutor.java

```diff
@@ -142,17 +146,45 @@ private ExecutionResult envNotFound(String environmentVariable) {
         return ExecutionResult.failure(message);
     }
 
-    private void setMetadata(GoEnvironment env, String bucket, S3ArtifactStore store) {
+    private void setMetadata(GoEnvironment env, String bucket, String destinationPrefix, S3ArtifactStore store) {
         ObjectMetadata metadata = metadata(env);
         metadata.setContentLength(0);
         InputStream emptyContent = new ByteArrayInputStream(new byte[0]);
         PutObjectRequest putObjectRequest = new PutObjectRequest(bucket,
-                env.artifactsLocationTemplate() + "/",
+                ensureKeySegmentValid(destinationPrefix),
                 emptyContent,
                 metadata);
 
         store.put(putObjectRequest);
     }
+
+    private String getDestinationPrefix(final TaskConfig config, final GoEnvironment env) {
+        String destinationPrefix = config.getValue(DESTINATION_PREFIX);
+
+        if(org.apache.commons.lang3.StringUtils.isBlank(destinationPrefix)) {
```

Comment:
Done. I changed to to not set the metadata if a destinationPrefix is configured. I chose this instead of adding another setting as most using a destinationPrefix are deploying rather than storing artifacts.